ecommerce/extensions/api/v2/views/coupons.py

```
@@ -241,6 +248,11 @@ def update(self, request, *args, **kwargs):
             # Remove catalog if switching from single course to dynamic query
             if voucher_range.catalog:
                 range_data['catalog'] = None
+
+            course_catalog_data = request.data.get('course_catalog')
+            if course_catalog_data:
+                range_data['course_catalog'] = course_catalog_data.get('id')
```

I don't think we need same logic here, since on coupon creation we are doing all stuff in `transaction.atomic` so any error will roll back the transaction. But here we are only updating individual fields so it would make sense to let the exception raise and this is the current behavior too.